### ISSUE TITLE
Extending an existing service?

### ISSUE THREAD

**Issue by jwvanderbeck** [krpc/krpc#305]
Is it possible to make a kRPC extension service that adds functionality onto existing services?

For example, say I want to add a new method to  Vessel.Parts that returns a list of parts based on something in my own extended service?  For example for TestFlight I would like to do something like active_vessel.parts.with_failure("blah") or active_vessel.parts.failed

**Comment by jwvanderbeck** [comment]
Also, I am not sure I understand the difference between a KRPCMethod and a KRPCProcedure aside from one being static and the other not.

**Comment by dewiniaid** [comment]
Probably related to #143.  My [proposed implementation there](https://github.com/krpc/krpc/issues/143#issuecomment-231614425) would handle the extension bits fairly nicely.

**Comment by djungelorm** [comment]
@jwvanderbeck yep that's the only difference between them. KRPCMethod is used on non-static methods of a KRPCClass and KRPCProcedure is used on static methods in a KRPCService.

**Comment by djungelorm** [comment]
The C# way to extend an existing class is to use extension methods. One way to do this would be to write an extension method, and annotate it with something like `KRPCExtensionMethod` so that the scanner picks it up and adds an RPC for it.

I think doing it this way would be a fairly easy addition.
